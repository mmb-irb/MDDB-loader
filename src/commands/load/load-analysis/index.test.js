const { loadAnalysis } = require(__dirname);

const analyses = [['rgyr', 'rgyr'], ['rmsd', 'rmsd'], ['rmsf', 'fluctuation']];

describe('loadAnalysis', () => {
  for (const [type, name] of analyses) {
    test(name, async () => {
      const analysis = await loadAnalysis(
        `${__dirname}/__fixtures/`,
        `md.${type}.xvg`,
      );
      expect(analysis.value).toMatchSnapshot();
    });
  }
  test('unsupported analysis', () => {
    expect(
      loadAnalysis(`${__dirname}/__fixtures/`, 'md.unsupported.xvg'),
    ).resolves.toBeUndefined();
  });
});
