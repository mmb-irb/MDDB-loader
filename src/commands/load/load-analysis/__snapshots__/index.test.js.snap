// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadAnalysis fluctuation 1`] = `
Object {
  "start": 1,
  "step": 1,
  "y": Object {
    "rmsf": Object {
      "average": 0.6547700000000001,
      "data": Array [
        0.5952,
        0.6155,
        0.6124,
        0.6155,
        0.5537,
        0.5615,
        0.6393,
        0.6985,
        0.7896,
        0.8665,
      ],
      "max": 0.8665,
      "min": 0.5537,
      "stddev": 0.10138094331118974,
    },
  },
}
`;

exports[`loadAnalysis rgyr 1`] = `
Object {
  "start": 0,
  "step": 10,
  "y": Object {
    "rgyr": Object {
      "average": 2.666339090909091,
      "data": Array [
        2.6359,
        2.66498,
        2.66173,
        2.67624,
        2.67035,
        2.67217,
        2.6647,
        2.67801,
        2.67116,
        2.66293,
        2.67156,
      ],
      "max": 2.67801,
      "min": 2.6359,
      "stddev": 0.011396322612619813,
    },
    "rgyrx": Object {
      "average": 2.3192290909090905,
      "data": Array [
        2.2906,
        2.31907,
        2.31611,
        2.32778,
        2.32038,
        2.32491,
        2.32012,
        2.33074,
        2.32249,
        2.31873,
        2.32059,
      ],
      "max": 2.33074,
      "min": 2.2906,
      "stddev": 0.010402517440067566,
    },
    "rgyry": Object {
      "average": 2.1215263636363635,
      "data": Array [
        2.10045,
        2.11922,
        2.12208,
        2.13055,
        2.12674,
        2.12122,
        2.11672,
        2.1304,
        2.12234,
        2.12384,
        2.12323,
      ],
      "max": 2.13055,
      "min": 2.10045,
      "stddev": 0.008181079724250662,
    },
    "rgyrz": Object {
      "average": 2.0830290909090907,
      "data": Array [
        2.05846,
        2.0821,
        2.07413,
        2.08966,
        2.0867,
        2.09195,
        2.08273,
        2.09106,
        2.09091,
        2.07248,
        2.09314,
      ],
      "max": 2.09314,
      "min": 2.05846,
      "stddev": 0.010774654940688763,
    },
  },
}
`;

exports[`loadAnalysis rmsd 1`] = `
Object {
  "start": 0,
  "step": 10,
  "y": Object {
    "rmsd": Object {
      "average": 0.12929449999999998,
      "data": Array [
        0.1963333,
        0.1727465,
        0.1555406,
        0.1403201,
        0.1393364,
        0.1345313,
        0.1372634,
        0.1293104,
        0.1127455,
        0.1041107,
        0.0000013,
      ],
      "max": 0.1963333,
      "min": 0.0000013,
      "stddev": 0.05000928996652521,
    },
  },
}
`;
